Source/WebKit/UIProcess/mac/WKSharingServicePickerDelegate.mm

```diff
@@ -151,21 +151,23 @@ - (void)sharingService:(NSSharingService *)sharingService didShareItems:(NSArray
         WeakPtr weakPage = menuProxy ? menuProxy->page() : nullptr;
         RetainPtr<NSString> itemUTI = itemProvider.get().registeredTypeIdentifiers.firstObject;
         [itemProvider loadDataRepresentationForTypeIdentifier:itemUTI.get() completionHandler:[weakPage, attachmentID = _attachmentID, itemUTI](NSData *data, NSError *error) {
-            RefPtr webPage = weakPage.get();
-            
-            if (!webPage)
-                return;
-            
-            if (error)
-                return;
-            
-            auto apiAttachment = webPage->attachmentForIdentifier(attachmentID);
-            if (!apiAttachment)
-                return;
-            
-            RetainPtr attachment = wrapper(apiAttachment);
-            [attachment setData:data newContentType:itemUTI.get()];
-            webPage->didInvalidateDataForAttachment(*apiAttachment.get());
+            ensureOnMainRunLoop([weakPage = WTFMove(weakPage), attachmentID, itemUTI, data = RetainPtr { data }, error = RetainPtr { error }] {
+                RefPtr webPage = weakPage.get();
+
+                if (!webPage)
+                    return;
+
+                if (error)
```

Comment:
Seems a little unfortunate to dispatch to the background thread just to early return. I recommend we null check the error on the background thread and early return there. Then we don't need to capture `error` in the lambda.